### PR TITLE
Deprecate `cairo_project.toml` in favour of `Scarb.toml`

### LINKED ISSUE #1737: Deprecate `cairo_project.toml` in favour of `Scarb.toml`
We should deprecate `cairo_project.toml` and use `Scarb.toml` instead.
The most reasonable way to do it is change the format of the argument `linked_libraries`/`cairo_path` in Protostar and related  `maybe_cairo_paths` in the compiler + change the way `ProjectConfig` is built in the compiler.

TODO in the same PR:
- [x] #1815
- [x] #1898

### LINKED ISSUE #1815: Deprecate `--linked-libraries`
After we have a proper support for `Scarb.toml` we can deprecate this argument

### LINKED ISSUE #1898: Replace `cairo_project.toml` with `Scarb.toml` in templates and tests
It will be a must after deprecating `cairo_project.toml`.
Blocked by #1737

### PR DESCRIPTION
**Pull request by piotmag769**
Issues for docs, error messages and tests:
https://github.com/software-mansion/protostar/issues/1954 
https://github.com/software-mansion/protostar/issues/1957

### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [ ] Added relevant tests
- [x] checklist from #1737

### CODE REVIEW

**Comment by piotmag769** [comment]
Merge asap to avoid conflicts with rust rewrites.
Docs + additional tests can be done in another PR

File: protostar/testing/test_scheduler.py
Comment (MaksymilianDemitraszek) [inline protostar/testing/test_scheduler.py:46]: [[Nitpick]] Unrelated change such changes make code harder to review

File: protostar/testing/test_scheduler.py
Comment (MaksymilianDemitraszek) [inline protostar/testing/test_scheduler.py:62]: [[Nitpick]] Unrelated change, such changes make code harder to review

File: protostar/commands/cairo1_commands/test_command.py
Comment (MaksymilianDemitraszek) [inline protostar/commands/cairo1_commands/test_command.py:139]: Thos should be also used in hint locals

File: protostar/contract_path_resolver/contract_path_resolver.py
Comment (MaksymilianDemitraszek) [inline protostar/contract_path_resolver/contract_path_resolver.py:29]: We shouldn't expose this one, but probably it won't be necessary if we don't call scarb metadata in hint locals

**Comment by MaksymilianDemitraszek** [comment]
Remember about updating default template and docs (in separate PRs)

File: .circleci/config.yml
Comment (piotmag769) [reply .circleci/config.yml:23]: Not really, but again - is it worth to create a new PR just for that? We will just forget to do it in the future

File: protostar/commands/cairo1_commands/fetch_from_scarb.py
Comment (piotmag769) [reply protostar/commands/cairo1_commands/fetch_from_scarb.py:28]: This one is on purpose, check TODO list https://github.com/software-mansion/protostar/issues/1957

File: .circleci/config.yml
Comment (MaksymilianDemitraszek) [reply .circleci/config.yml:23]: For such things i think it is generally worth it imo, you don't have to wait until PR is merged to fix it. You can leave it for this one though

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**
